packages/relay/src/lib/config/hbarSpendingPlanConfigService.ts

```diff
@@ -0,0 +1,315 @@
+/*
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import fs from 'fs';
+import findConfig from 'find-config';
+import { isValidSpendingPlanConfig, SpendingPlanConfig } from '../types/spendingPlanConfig';
+import { HbarSpendingPlanRepository } from '../db/repositories/hbarLimiter/hbarSpendingPlanRepository';
+import { EthAddressHbarSpendingPlanRepository } from '../db/repositories/hbarLimiter/ethAddressHbarSpendingPlanRepository';
+import { IPAddressHbarSpendingPlanRepository } from '../db/repositories/hbarLimiter/ipAddressHbarSpendingPlanRepository';
+import { RequestDetails } from '../types';
+import { Logger } from 'pino';
+import { SubscriptionType } from '../db/types/hbarLimiter/subscriptionType';
+import { IDetailedHbarSpendingPlan } from '../db/types/hbarLimiter/hbarSpendingPlan';
+
+export class HbarSpendingPlanConfigService {
+  private readonly TTL = -1; // -1 means no TTL, i.e. the data will not expire
+  private readonly DEFAULT_SPENDING_PLANS_CONFIG_FILE = 'spendingPlansConfig.json';
```

Comment:
Since `DEFAULT_SPENDING_PLANS_CONFIG_FILE` is only used once for the filename in `loadSpendingPlansConfig()`, can we simplify it to something like:

```suggestion
  private readonly TTL = -1; // -1 means no TTL, i.e. the data will not expire
  private readonly SPENDING_PLANS_CONFIG_FILE = process.env.HBAR_SPENDING_PLANS_CONFIG_FILE || 'spendingPlansConfig.json';
```
Then instead of initializing a new filename variable in `loadSpendingPlansConfig()`, we can simply reuse `this.SPENDING_PLANS_CONFIG_FILE`.

Just a suggestion I'm fine with current flow.